middlewares/user/getTargetUserIdByPMChannelId.js

```diff
@@ -0,0 +1,57 @@
+const {StreamChat} = require('stream-chat');
+const Getstream = require('../../vendor/getstream');
+
+/**
+ *
+ * @param {import('express').Request} req
+ * @param {import('express').Response} res
+ * @param {import('express').NextFunction} next
+ * @returns
+ */
+const getTargetUserIdByPMChannelIdMiddleware = async (req, res, next) => {
+  const {userId} = req || {};
+  const {channelId} = req?.query || {};
+
+  if (!channelId) {
+    return res.status(400).json({
+      code: 400,
+      success: false,
+      message: 'Channel ID is required'
+    });
+  }
+
+  const chatClient = new StreamChat(process.env.API_KEY, process.env.SECRET);
+  let channelMembers;
+  try {
+    let {users} = await Getstream.chat.getTargetUserIdByChannelId(chatClient, userId, channelId);
+    channelMembers = users;
+  } catch (e) {
+    console.error('Failed to get channel members', e);
+    return res.status(500).json({
+      code: 500,
+      success: false,
+      message: 'Internal server error'
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling here only logs the error message and returns a generic 'Internal server error' message. It would be more helpful to return the actual error message to the client, so they can understand what went wrong. Also, sensitive information like API keys should not be logged.
```diff
- console.error('Failed to get channel members', e);
+ console.error('Failed to get channel members');
  return res.status(500).json({
-   code: 500,
-   success: false,
-   message: 'Internal server error'
+   code: e.code || 500,
+   success: false,
+   message: e.message || 'Internal server error'
  });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->